pydantic/_internal/_fields.py

```diff
@@ -226,6 +226,17 @@ def collect_model_fields(  # noqa: C901
                     # Nothing stops us from just creating a new FieldInfo for this type hint, so we do this.
                     field_info = FieldInfo.from_annotation(ann_type)
         else:
+            if hasattr(ann_type, '__args__'):
+                for anno_arg in ann_type.__args__:
+                    if _typing_extra.is_annotated(anno_arg):
+                        for anno_type_arg in _typing_extra.get_args(anno_arg):
+                            if isinstance(anno_type_arg, FieldInfo) and anno_type_arg.alias is not None:
+                                warnings.warn(
+                                    f'Field "{ann_name}" has `alias` should be set at higher level to have affect',
+                                    UserWarning,
+                                )
+                                break
+                        break
```

Comment:
```suggestion
            for anno_arg in getattr(ann_type, '__args__', ()):
                if not _typing_extra.is_annotated(anno_arg):
                    continue
                for anno_type_arg in _typing_extra.get_args(anno_arg):
                    if isinstance(anno_type_arg, FieldInfo) and anno_type_arg.alias is not None:
                        warnings.warn(
                            f'Field "{ann_name}" has `alias` should be set at higher level to have affect',
                            UserWarning,
                        )
                        break
```
Note: I removed the second break, and it seems to me like including it is problematic — if there are multiple annotated args it looks to me like it won't hit them.

I would prefer if we broke this out into a separate function (private in this module) to get this check moved out-of-line instead of making this method bigger and more complicated than it already is. (In that case, the `break` can just become `return`)

Reply:
@dmontagu Thank for the suggestions.
As my understand and please correct me if i'm wrong, warning on the first annotated args that has `alias` set or on all annotated args that have it is the same things !?.

Reply:
I think we want to warn for each argument where it's a problem, which is what @dmontagu is pointing out.

I agree with his suggestions above! Feel free to ping me when you've made the changes, and I can do a second review!

Reply:
@nix010,

Let me know if you have any questions about this suggestion. Thanks for your work on this!

Reply:
@sydney-runkle Made the changes, please review. Lost my laptop in a fire accident so it took a bit longer than usual.

Reply:
Resolving as fixed.